src/components/admin/hazardous-materials-specifiers.tsx

```diff
@@ -182,54 +220,83 @@ const HazardousMaterialsSpecifiers: React.FC = () => {
+            strings.formatString(strings.adminScreen.addNewWasteSpecifierDialog.text.en)
+          }
+          onChange={ handleNewLocalizedNameChange }
+        />
+      ))}
     </GenericDialog>
   );
 
   /**
    * Renders delete hazardous waste specifier dialog
    */
-  const renderDeleteHazardousWasteSpecifierDialog = () => (
-    <GenericDialog
-      error={ false }
-      open={ deletingHazardousWasteSpecifier }
-      onClose={ () => setDeletingHazardousWasteSpecifier(false) }
-      onCancel={ () => setDeletingHazardousWasteSpecifier(false) }
-      onConfirm={ onDeleteHazardousWasteSpecifierConfirm }
-      title={ strings.adminScreen.deleteWasteSpecifierDialog.title }
-      positiveButtonText={ strings.generic.confirm }
-      cancelButtonText={ strings.generic.cancel }
-    >
-      <Typography>
-        { strings.formatString(strings.adminScreen.deleteWasteSpecifierDialog.text, deletableHazardousWasteSpecifier ? deletableHazardousWasteSpecifier.name : "") }
-      </Typography>
-    </GenericDialog>
-  );
+  const renderDeleteHazardousWasteSpecifierDialog = () => {
+    if (!deletableHazardousWasteSpecifier) {
+      return null;
+    }
+
+    return (
+      <GenericDialog
+        error={ false }
+        open={ deletingHazardousWasteSpecifier }
+        onClose={ () => setDeletingHazardousWasteSpecifier(false) }
+        onCancel={ () => setDeletingHazardousWasteSpecifier(false) }
+        onConfirm={ onDeleteHazardousWasteSpecifierConfirm }
+        title={ strings.adminScreen.deleteWasteSpecifierDialog.title }
+        positiveButtonText={ strings.generic.confirm }
+        cancelButtonText={ strings.generic.cancel }
+      >
+        <Typography>
+          {
+            strings.formatString(strings.adminScreen.deleteWasteSpecifierDialog.text,
+              LocalizationUtils.getLocalizedName(deletableHazardousWasteSpecifier.localizedNames, selectedLanguage))
+          }
+        </Typography>
+      </GenericDialog>
+    );
+  };
 
   /**
    * Renders edit hazardous waste specifier dialog
    */
-  const renderEditHazardousWasteSpecifierDialog = () => (
-    <GenericDialog
-      error={ false }
-      open={ editingHazardousWasteSpecifier }
-      onClose={ () => setEditingHazardousWasteSpecifier(false) }
-      onCancel={ () => setEditingHazardousWasteSpecifier(false) }
-      onConfirm={ onEditReusableMaterialConfirm }
-      title={ strings.adminScreen.updateWasteSpecifierDialog.title }
-      positiveButtonText={ strings.generic.confirm }
-      cancelButtonText={ strings.generic.cancel }
-    >
-      <TextField
-        label={ strings.adminScreen.updateWasteSpecifierDialog.text }
-        value={ editableHazardousWasteSpecifier?.name }
-        onChange={ handleEditableNameChange }
-      />
-    </GenericDialog>
-  );
+  const renderEditHazardousWasteSpecifierDialog = () => {
+    if (!editableHazardousWasteSpecifier) {
+      return null;
+    }
+
+    return (
+      <GenericDialog
+        error={ false }
```

Comment:
This should be optional property, so that it would not be required to be defined at all.